File: practica2/c++/write.cc
#include "write.h"
#include <iostream>
#include <cstdint>

using namespace std;

void write::ejecutar(stack<int32_t>& pila, int32_t& pc) const
{
    int32_t cima = pila.top();
    pila.pop();
    cout << cima << endl;
    pc++;
}		

string write::nombre() const
{
    return "write";
}

File: practica2/java/Write.java
import java.util.Stack;

/**
 * write
 */
public class Write extends Inst{

	@Override
    public int ejecutar(Stack<Integer> pila, int pc){
        int cima = pila.pop();
        System.out.println(cima);
        pc++;
        return pc;
    }

	@Override
    public String nombre(){
        return "write";
    }
    
}
